### ISSUE TITLE
Chunks look corrupted on map

### ISSUE THREAD

**Issue by lhns** [LolHens/dynmap-fabric#19]
![image](https://user-images.githubusercontent.com/1524059/90264037-c90c5500-de50-11ea-9313-7f3031ea56e5.png)
![image](https://user-images.githubusercontent.com/1524059/90266752-c4e23680-de54-11ea-88f2-2c946850531b.png)

This has probably something to do with #9 because I see a lot of warnings in the server log:
```
14T15:09:08.615519882Z [17:09:08] [Dynmap Render Thread/WARN]: [Dynmap] Got statelist of length 320 but expected a length of 342
```

~~Maybe these are chunks which are not fully generated yet?~~
~~This could also be a weird mod interaction because I am not seeing this on my "vanilla" server.~~
Affected versions: 1.16.1, 1.16.2
Also a rerender of the affected chunks seems to fix this.

**Comment by Lemonszz** [comment]
Getting the same thing over on my server.

It's mostly 320/342, but sometimes other numbers like 384/410.
I don't believe these chunks are still generating as this is a fairly old and established world.
It's not a mod interaction, 1.16.2 Fabric server with no other mods installed (Only dynmap & FAPI).

It's happened in areas that have been modified in the past (http://lemons.party/i/u54pr.png) as well as areas that may not have been modified (http://lemons.party/i/u1wvx.png).

Most areas seem intact. 

Let me know if you want more info.

```
[09:24:05] [Dynmap Render Thread/WARN]: [Dynmap] Got statelist of length 320 but expected a length of 342
[09:24:05] [Dynmap Render Thread/WARN]: [Dynmap] Got statelist of length 320 but expected a length of 342
[09:24:06] [Dynmap Render Thread/WARN]: [Dynmap] Got statelist of length 320 but expected a length of 342
[09:24:06] [Dynmap Render Thread/WARN]: [Dynmap] Got statelist of length 320 but expected a length of 342
[09:24:06] [Dynmap Render Thread/WARN]: [Dynmap] Got statelist of length 320 but expected a length of 342
[09:24:06] [Dynmap Render Thread/WARN]: [Dynmap] Got statelist of length 384 but expected a length of 410
[09:24:06] [Dynmap Render Thread/WARN]: [Dynmap] Got statelist of length 320 but expected a length of 342
[09:24:06] [Dynmap Render Thread/WARN]: [Dynmap] Got statelist of length 320 but expected a length of 342
[09:24:06] [Dynmap Render Thread/WARN]: [Dynmap] Got statelist of length 320 but expected a length of 342
[09:24:06] [Dynmap Render Thread/WARN]: [Dynmap] Got statelist of length 384 but expected a length of 410
[09:24:06] [Dynmap Render Thread/WARN]: [Dynmap] Got statelist of length 320 but expected a length of 342
[09:24:06] [Dynmap Render Thread/WARN]: [Dynmap] Got statelist of length 320 but expected a length of 342
[09:24:12] [Dynmap Render Thread/WARN]: [Dynmap] Got statelist of length 320 but expected a length of 342
[09:24:12] [Dynmap Render Thread/WARN]: [Dynmap] Got statelist of length 320 but expected a length of 342
[09:24:12] [Dynmap Render Thread/WARN]: [Dynmap] Got statelist of length 320 but expected a length of 342
[09:24:13] [Dynmap Render Thread/WARN]: [Dynmap] Got statelist of length 320 but expected a length of 342
```

**Comment by Lemonszz** [comment]
Can confirm visiting the area in game fixes it.
Before: http://lemons.party/i/ht52y.png
After: http://lemons.party/i/srwbd.png

**Comment by LarveyOfficial** [comment]
mmm, maybe the chunks aren't loaded / generated all the way? Any idea of why this is happening @LolHens ?

**Comment by lhns** [comment]
I looked through the code and this should never happen normally. Also I cannot reproduce this in my dev environment and on prod it only happens sporadically. So the search goes on...

**Comment by lhns** [comment]
tried to fix this via #22 and #23

**Comment by lhns** [comment]
Could you try out v0.2.11? It seems to have fixed it on my system. The warning still appears in the log but now it won't try to fill in the missing data but instead skip the chunk and hope that dynmap will render it again later. Seems to work pretty well.

**Comment by LarveyOfficial** [comment]
Ill give it a shot right now 👍

**Comment by LarveyOfficial** [comment]
Do you have a discord server for this project? I feel like it would help for testing.

**Comment by lhns** [comment]
Not until now: https://discord.gg/2SMQsHd

**Comment by lhns** [comment]
Okay I tested this a bit more with @LuisVervaet and we found out that this is a 1.16 bug on all platforms (spigot, forge, fabric).
Apparently there is even an issue on the official repo: https://github.com/webbukkit/dynmap/issues/3083
I will continue to investigate this but at least this is not a fabric only thing.

**Comment by lhns** [comment]
properly fixed in 0.12.2

**Comment by Lemonszz** [comment]
Yeah seems good I think.
